### DISCUSSION TITLE
How to get raw rich text editor content?

### DISCUSSION THREAD

**Discussion by benstigsen** [pocketbase/pocketbase#5786] [Q&A]
Currently if I enter some code like this into the rich text editor:
```
<p>Hello World!</p>
```

When I get the content later using JSVM, it results in:
```
<p>&lt;p&gt;Hello World!&lt;p&gt;</p>
```

I could use "plain text" for the content above, but as soon as it's multiple lines, that wouldn't be possible anymore.
TinyMCE has functionality to get the raw content, alternatively having a `plain text multiline` would also fix it. 

It's not possible to use `new DOMParser.parseString(richtext.content).innerText` or similar, due to there not being an actual DOM and the limitations of `goja`. Is there another way to do it?

**Comment by ganigeorgiev** [comment]
The string generated from the TinyMCE is stored and returned as it is.

I'm not sure what do you mean by _"get the content later using JSVM"_ but if you are trying to render the value as raw html in a Go template then you need to use the `raw` function (e.g. `{{.content|raw}}`).

**Reply by benstigsen** [reply]
Hmm, I'll try to elaborate. I have something like this:
![image](https://github.com/user-attachments/assets/a76c97a3-8566-41ec-be15-8f9c12f36ece)

Then I have JS like this:
```js
const sections = $app
  .dao()
  .findRecordsByExpr("sections")
  ?.map((section) => section.publicExport());

for (const section of sections) {
  // get the template referred to by each "section"
  // relative single to the "templates" collection
  const sectionTemplate = $app
    .dao()
    .findRecordById("templates", section.sectionTemplate)
    .publicExport();

  // print rich text editor content
  console.log(sectionTemplate.content);
}
```

This prints the following:
```
<p>&lt;!doctype html&gt;<br>&lt;html lang="en"&gt;<br>&nbsp; &lt;head&gt;<br>&nbsp; &nbsp; {{ template "head" }}<br>&nbsp; &nbsp; &lt;meta name="description" content="{{ .section.description }}" /&gt;<br>&nbsp; &nbsp; &lt;meta name="keywords" content="{{ .section.keywords }}" /&gt;<br>&nbsp; &nbsp; &lt;title&gt;{{ .section.title }}&lt;/title&gt;<br>... (and more like this)
```

Ideally I'd like to get the _actual_ content, without formatting so that I get `<` instead of `&lt;` and without it being wrapped in further tags. So like in the screenshot I'd like to get something like:

```html
<!doctype html>
<html lang="en">
  <head>
    {{ template "head" }}
    <title>Benjamin Stigsen</title>
    ... and so on
```

Right now I'm trying to use `$template.loadString()` on the content, but because everything is wrapped in tags by TinyMCE to format stuff and due to the limitations of Goja, it's not possible for me to get the content without formatting.

**Reply by ganigeorgiev** [reply]
That's not how the `editor` is expected to be used. When you are typing `<p>` in the editor, TinyMCE automatically escapes the HTML entities (you can click on the "Source code" button in the editor to see what will be sent and stored on the server).

If you want to be able to type HTML tags without escaping use a plain `text` field.

**Reply by benstigsen** [reply]
> If you want to be able to type HTML tags without escaping use a plain text field.

Yeah but the plain text field only allows for a single line, not multiline input.

**Reply by ganigeorgiev** [reply]
No, it supports multiple lines (you can type them with "Shift + Enter").

**Reply by benstigsen** [reply]
Ah that works, though the behaviour is weird as it scrolls the page up to the first line of the plain text field, if one creates a newline.
